src/Tests/Microsoft.Diagnostics.Monitoring.TestCommon/AssemblyHelper.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Reflection;
+
+namespace Microsoft.Diagnostics.Monitoring.TestCommon
+{
+    public static class AssemblyHelper
+    {
+        public static string GetAssemblyArtifactBinPath(
+            Assembly testAssembly,
+            string assemblyName,
+            TargetFrameworkMoniker tfm = TargetFrameworkMoniker.Current)
+        {
+            string assemblyPath = testAssembly.Location
+                .Replace(testAssembly.GetName().Name, assemblyName);
+
+            if (tfm != TargetFrameworkMoniker.Current)
+            {
+                string currentFolderName = GetTargetFrameworkMonikerFolderName(DotNetHost.BuiltTargetFrameworkMoniker);
+                string targetFolderName = GetTargetFrameworkMonikerFolderName(tfm);
+
+                assemblyPath = assemblyPath.Replace(currentFolderName, targetFolderName);
+            }
+
+            return assemblyPath;
+        }
+
+        private static string GetTargetFrameworkMonikerFolderName(TargetFrameworkMoniker moniker)
+        {
+            switch (moniker)
+            {
+                case TargetFrameworkMoniker.Net50:
+                    return "net5.0";
+                case TargetFrameworkMoniker.Net60:
+                    return "net6.0";
```

Comment:
These are the same as `$(TargetFramework)` in the `.targets` files. Why not generate these into an extension method in `DotNetHost.cs.template` and keep all the re-configuration necessary for a version update in the same place?

Reply:
I've moved this conversion logic to be an extension method in the TargetFrameworkMoniker.cs file.